packages/devtools_app/lib/src/memory/memory_heap_tree_view.dart

```diff
@@ -339,11 +344,11 @@ class HeapTreeViewState extends State<HeapTree> with AutoDisposeMixin {
           children: [
             const Text('Heat Map'),
             Switch(
-              value: controller.showHeatMap,
+              value: controller.showHeatMap.value,
```

Comment:
I'm not seeing where we are listening for changes to `controller.showHeatMap` to update this UI.